mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMavericksViewModel.kt

```diff
@@ -389,10 +441,9 @@ abstract class BaseMavericksViewModel<S : MvRxState>(
         } else {
             flowWhenStarted(lifecycleOwner)
         }
-        val scope = lifecycleOwner?.lifecycleScope ?: viewModelScope
-        return scope.launch {
-            flow.collectLatest { action(it) }
-        }
+        return viewModelScope.launch {
+            flow.collectLatest(action)
+        }.cancelOnClear(lifecycleOwner?.lifecycleScope)
```

Comment:
Is there a reason for this change (comparing to just launch in `lifecycleOwner?.lifecycleScope ?: viewModelScope`)? Previous version looks more idiomatic and also behavior differs

Reply:
@denis-bezrukov Yeah, before [this test](https://github.com/airbnb/MvRx/blob/master/mvrx/src/test/kotlin/com/airbnb/mvrx/InterViewModelSubscriberTest.kt#L251) was broken.

The tricky thing here is that we don't know which scope has a shorter lifecycle so we have to cancel on whichever is cancelled first.

Reply:
@gpeal as I see that just means we don't close CoroutineStateStore.stateChannel properly. So what do you think about just closing it when store's scope cancelled?
Then you will be able to subscribe here as previously. 

It seems that it will provide better expectations and cleaner implementation.

Reply:
we need to make flush as:
```
try {
    while (isActive) {
        flushQueuesOnce()
    }
} catch (e: Exception) {
    stateChannel.cancel(e as? CancellationException)
    throw e
}
```

Reply:
@denis-bezrukov. Good point. Updated